web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -493,14 +667,333 @@ $.fn.drawer = function(id, ajax) {
+            // view. With #sub_map_links detached from the DOM, we set up the
+            // individual report's sub_map_links using map_controls().
+            if (!('originalSubMapLinks' in window)) {
+                window.originalSubMapLinks = $('#sub_map_links').detach();
+            }
+            fixmystreet.set_up.map_controls();
+
+            $sideReport.find('#key-tool-problems-nearby').addClass('.js-back-to-report-list');
+            fixmystreet.set_up.map_sidebar_key_tools();
+
+            fixmystreet.set_up.fancybox_images();
+            fixmystreet.set_up.dropzone($sideReport);
+            fixmystreet.set_up.form_focus_triggers();
+
+            window.selected_problem_id = reportId;
+            if ($('html').hasClass('mobile') && fixmystreet.map.panTo) {
+                fixmystreet.map.panTo(
+                    fixmystreet.maps.get_marker_by_id(reportId).geometry.getBounds().getCenterLonLat()
+                );
+            }
+            if (fixmystreet.markers.refresh) {
+                fixmystreet.markers.refresh({force: true}); // force a redraw so the selected marker gets bigger
+            }
+
+            if (typeof callback === 'function') {
+                callback();
+            }
+
+        } else {
+            window.location.href = reportPageUrl;
+        }
+
+    }).fail(function(jqXHR, textStatus, errorThrown) {
+        window.location.href = reportPageUrl;
+
+    });
+  },
+
+  around: function(reportListUrl, callback) {
+    // If the report list is already in the DOM,
+    // just reveal it, rather than loading new page.
+    if ($('#side').length) {
+        $('#side').show();
+        $('#side-form').hide();
+        $('#side-report').remove();
+
+        $('body').removeClass('with-notes');
+
+        document.title = fixmystreet.original_title;
+        fixmystreet.page = 'around';
+        if ($('html').hasClass('mobile')) {
+            $('#mob_sub_map_links').remove();
+            fixmystreet.mobile_reporting.apply_ui();
+        }
+
+        if ('originalSubMapLinks' in window) {
+            $('#sub_map_links').replaceWith(window.originalSubMapLinks);
+            delete window.originalSubMapLinks;
+        }
+        $('#sub_map_links').show();
+        fixmystreet.set_up.map_controls();
+
+        window.selected_problem_id = undefined;
+
+        // Perform vendor-specific map setup steps,
+        // to get map back into "around" mode.
+        fixmystreet.maps.display_around();
+
+        if (typeof callback === 'function') {
+            callback();
+        }
+    } else {
+        window.location.href = reportListUrl;
+    }
+  }
+};
+
+
+$(function() {
+    window.cobrand = $('meta[name="cobrand"]').attr('content');
+    fixmystreet.original_title = document.title;
+
+    if (typeof variation !== 'undefined' && variation === 1) {
+        $('input[name=variant]').val(1);
+    }
+
+    $.each(fixmystreet.set_up, function(key, value) {
```

Comment:
Tiny suggestion: Use more descriptive variables here? Like `(setup_name, setup_func)` or something? If nothing else, it'll make `value()` on the next line look a lot less mysterious.